File: src/java/magma/node/CDefinition.java
package magma.node;

import magma.option.None;
import magma.option.Option;

public record CDefinition(Option<String> maybeTypeParameter, String type, String name) implements JavaMethodHeader {
	public CDefinition(final String type, final String name) {
		this(new None<>(), type, name);
	}

	@Override
	public String generate() {
		return this.maybeTypeParameter.map(value -> "<" + value + "> ").orElse("") + this.type + " " + this.name;
	}
}

File: src/windows/magma/node/CDefinition.c
#include "CDefinition.h"
#include "../../java/util/function/Function.h"
/*

public record CDefinition(String beforeType, CType type, String name) implements Header {
    @Override
    public String generate() {
        return Placeholder.generate(this.beforeType() + " ") + this.type()
                .generate() + " " + this.name();
    }

    public CDefinition mapName(final Function<String, String> mapper) {
        return new CDefinition(this.beforeType, this.type, mapper.apply(this.name));
    }
}*//**/